gossip/api/types.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
```

Comment:
The `types.go` file contains two things:
- `*Request` and `*Response` structs. 
- Definitions for TLS encodings. 

It might be worth separating these two things into separate go packages to ease the transition to a gRPC based API.  There should be a 1:1 mapping between the Request and Response structs in proto3.  Is there anything there that you think would cause a problem?  

Encoding the parameters of a request seems like a small, but separate? step / layer prior to composing the actual JSON/Protobuf request to send on the wire. 

Do you think that should be part of this PR or a follow up?

Reply:
I think splitting into separate packages would be a bad idea -- this package holds the entire API definition, so that everything a client needs to know about is in one place (including non-Go clients: they can't use the structures here, but `types.go` tells them everything they need to know to interact with the Hub).   It's also less than 100 lines of code, so splitting seems excessive.